lib/services/ios-project-service.ts

```diff
@@ -379,5 +393,57 @@ class IOSProjectService extends projectServiceBaseLib.PlatformProjectServiceBase
 			this.$fs.rename(path.join(fileRootLocation, oldFileName), path.join(fileRootLocation, newFileName)).wait();
 		}).future<void>()();
 	}
+	
+	private prepareDynamicFrameworks(pluginPlatformsFolderPath: string, pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			_.each(this.getAllDynamicFrameworksForPlugin(pluginData).wait(), fileName => this.addLibrary(path.join(pluginPlatformsFolderPath, fileName)).wait());			
+		}).future<void>()();
+	}
+	
+	private prepareCocoapods(pluginPlatformsFolderPath: string): IFuture<void> {
+		return (() => {
+			let pluginPodFilePath = path.join(pluginPlatformsFolderPath, "Podfile");
+			if(this.$fs.exists(pluginPodFilePath).wait()) {
+				let pluginPodFileContent = this.$fs.readText(pluginPodFilePath).wait();
+				let contentToWrite = this.buildPodfileContent(pluginPodFilePath, pluginPodFileContent);
+				this.$fs.appendFile(this.projectPodFilePath, contentToWrite).wait();
+			}
+		}).future<void>()();
+	}
+	
+	private removeDynamicFrameworks(pluginPlatformsFolderPath: string, pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			let project = this.createPbxProj();
+						
+			_.each(this.getAllDynamicFrameworksForPlugin(pluginData).wait(), fileName => {
+				let fullFrameworkPath = path.join(pluginPlatformsFolderPath, fileName);
+				let relativeFrameworkPath = this.getFrameworkRelativePath(fullFrameworkPath);
+				project.removeFramework(relativeFrameworkPath, { customFramework: true, embed: true })
+			});
+			
+			this.savePbxProj(project).wait();
+		}).future<void>()();
+	}
+	
+	private removeCocoapods(pluginPlatformsFolderPath: string): IFuture<void> {
+		return (() => {
+			let pluginPodFilePath = path.join(pluginPlatformsFolderPath, "Podfile");
+			if(this.$fs.exists(pluginPodFilePath).wait()) {
+				let pluginPodFileContent = this.$fs.readText(pluginPodFilePath).wait();
+				let projectPodFileContent = this.$fs.readText(this.projectPodFilePath).wait();
+				let contentToRemove= this.buildPodfileContent(pluginPodFilePath, pluginPodFileContent);
+				projectPodFileContent = helpers.stringReplaceAll(projectPodFileContent, contentToRemove, "");
+				if(_.isEmpty(projectPodFileContent)) {
```

Comment:
You may trim the content - there is no need to keep files consisting of empty space. This may happen when the user manually adds empty space.